doc/specs/#605 - Search/spec.md

```diff
@@ -0,0 +1,101 @@
+2. The user can choose to search up or down by choosing up arrow or down arrow buttons. A blue border will appear on the chosen button. If the user does not click the arrows buttons, the default direction is up. 
+3. The user can choose to do case sensitive or insensitive match by checking a check box. 
+4. If the user click the "X" button or press Esc, the search stopped and the search box disappears. In phrase one we do not store any state. 
+5. Once the search box is closed (exiting search mode), the selection will still be there. This coincides with the current VS Code and cmd experience. To get rid of the selection, the user can just click other area of the window. 
+
+## UI/UX Design
+
+![SearchBox mockup](images/SearchBoxControl.png)
+
+Above is the SearchBoxControl. The two buttons with up/down arrows controls the search direction, a blue border will appear on the current selected direction. The checkbox, if checked, means that we are searching case-sensitivily. 
+
+![SearchBox mockup, arrow button clicked](images/SearchBoxUpSelected.png)
+
+The search box should be on the top right corner of the Terminal window. If the current tab is split into panes, each pane will have a individual searchbox. We need to avoid it blocking too much screen contents. 
+
+## Capabilities
+
+1. The user can search exact matched text in the text buffer of the Terminal Screen. 
+2. The user can choose to search case sensitively and insensitively. 
+3. The user can search up or down. 
+4. Found text will be selected. 
+5. The search will start from the active selected text (inclusive) if there is one, or where the cursor is. 
+5. The search will automatically go around when it reaches the starting point. 
+
+### Accessibility
+
+This feature could help multitaksing terminal users. 
+
+### Security
+
+This feature should not introduce any new security issues.
+
+### Reliability
+
+This feature enable users to search for text in the terminal input/output history. This is a widely-used feature in most editors and thus improve the reliability of Terminal once added. 
+
+### Compatibility
+
+This feature won't break existing features of Terminal.
+
+### Performance, Power, and Efficiency
+
+This feature only launches in need. It does not impact the performance of Terminal. 
+
+## Potential Issues
+
+1. The search bar should not block command line view. 
+2. Search should not block any program the Terminal is currently executing.
+
+## Future considerations
+
+In version 1, we want realize a case sensitive/insensitive exact text match. But we may consider the following features in version 2:
+
+1. Add "Find" button in dropdown menu to trigger search. 
+2. Search from all tabs. For Version 1 we just want to realize search within one tab. However, the community also requests search from all tabs. We put in our goals for Version 2. 
+3. Regular experssion match. 
+4. Search history.
+5. High-light while you type. 

```

Comment:
Another idea for future consideration: Emphasizing all the other matches in the buffer with an outline or something, similar to how sublime works:
![image](https://user-images.githubusercontent.com/18356694/69557540-6266a480-0f6c-11ea-995e-c8a34234b8a7.png)


Though, this probably needs some cooperation with the renderer to support rendering like this.

Reply:
Added in the Future considerations.